tasks/main.yml

```diff
@@ -18,21 +22,39 @@
   changed_when: false
   register: __podman_version_output
 
+- name: Set podman version
+  set_fact:
+    podman_version: "{{ (__podman_version_output.stdout.split())[2] }}"
+
 - name: Podman package version must be 4.2 or later
   fail:
     msg: >
-      podman package version {{ __podman_version }} is too old -
+      podman package version {{ podman_version }} is too old -
       must be 4.2 or later
-  when: __podman_version is version("4.2", "<")
-  vars:
-    __podman_version: "{{
-      (__podman_version_output.stdout.split())[2] }}"
+  when: podman_version is version("4.2", "<")
+
+- name: Podman package version must be 4.4 or later for quadlet, secrets
+  fail:
+    msg: >
+      podman package version {{ podman_version }} is too old -
+      must be 4.4 or later to use quadlet or secrets
+  when:
+    - (podman_quadlet_specs | length > 0) or (podman_secrets | length > 0)
+    - podman_version is version("4.4", "<")
+    - podman_fail_if_too_old | d(true)
+
+- name: Podman package version must be 4.4 or later for quadlet, secrets
```

Comment:
Assuming I'm not misreading the `when` clause here, it is the same as the previous one. Now, since the previous one is a `fail` then the flow will never reach this stage. So, if you want the second one to happen, you should reorder them.
Another suggestion is to combine them using `block` so that you can write the `when` clause only once.

Reply:
What I'm trying to do here is - if `podman_fail_if_too_old` is `True`, then the role will fail if podman is "too old".  This is the default case, and one that production users will hit if they try to use newer features with an older version of podman.  If `podman_fail_if_too_old` is `False`, then I want to just skip the rest of the play - this is useful for testing purposes, not something a production user would or should ever use.